cli/src/base.ts

```diff
@@ -75,4 +98,118 @@ export abstract class BaseCommand extends Command {
       return this.error(`Error accessing the API: ${err instanceof Error ? err.message : String(err)}`);
     }
   }
+
+  async getWorkspace(options: { allowCreate?: boolean } = {}) {
+    const xata = await this.getXataClient();
+    const workspaces = await xata.workspaces.getWorkspacesList();
+
+    if (workspaces.workspaces.length === 0) {
+      if (!options.allowCreate) {
+        return this.error('No workspaces found, please create one first');
+      }
+
+      const result = await prompts({
+        type: 'text',
+        name: 'name',
+        message: 'New workspace name'
+      });
+      const workspace = await xata.workspaces.createWorkspace({ name: result.name, slug: slugify(result.name) });
+      return workspace.id;
+    } else if (workspaces.workspaces.length === 1) {
+      const workspace = workspaces.workspaces[0].id;
+      this.log(`You only have a workspace, using it by default: ${workspace}"`);
+      return workspace;
+    }
+
+    const result = await prompts({
+      type: 'select',
+      name: 'workspace',
+      message: 'Select a workspace',
+      choices: workspaces.workspaces.map((workspace) => ({
+        title: workspace.name,
+        description: workspace.id,
+        value: workspace.id
+      }))
+    });
+
+    return String(result.workspace);
+  }
```

Comment:
The behavior for workspaces is copied from here https://github.com/xataio/cli/blob/main/cmd/init.go#L57

I guess typically people would have one workspace, because when you sign up we are going to create a workspace for you. And unless you want to collaborate somehow with other people you won't create more workspaces.

Happy to discuss this on Monday during our sync if you want.